react-file-uploads/server/index.js

```diff
@@ -0,0 +1,61 @@
+require('dotenv/config');
+const pg = require('pg');
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+const uploadsMiddleware = require('./uploads-middleware');
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/uploads', uploadsMiddleware, (req, res, next) => {
+  const { caption } = req.body;
+  if (!caption) {
+    throw new ClientError(400, 'caption is a required field');
+  }
+  // v--- my code ---v
+  //* - create a url for the image by combining '/images' with req.file.filename
+  const url = '/images/' + req.file.filename;
+  //* - insert the "caption" and "url" into the "images" table
+  const sql = `
+    insert into "images"("caption", "url")
+    values('${caption}', '${url}')
```

Comment:
Never directly concat user provided information into a query, you should be using SQL parameter `$1, $2`

Reply:
right right, i shoulda remembered that, thanks!